### PR TITLE
Create minimal Electron app

### PR DESCRIPTION
**Pull request by hachibu**
screen shot 2016-12-17 at 3 13 48 pm

Create minimal Electron app.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,8 @@
-# rasa
\ No newline at end of file
+# Development
```
Comment (marinhero) [inline README.md]: Thanks for adding this! 👏

File: bin/setup-osx [hunk 2]
```diff
@@ -1,4 +1,20 @@
 #!/usr/bin/env bash
 
+vim_install() {
+  local DIRECTORY="$HOME/.vim/bundle/$2"
+
+  if [ -d "$DIRECTORY" ]; then
```
Comment (marinhero) [inline bin/setup-osx]: Cool 😎

File: src/Main.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
+import { app, BrowserWindow } from "electron";
```
Comment (marinhero) [inline src/Main.ts]: Wow! I can't believe this file can render a desktop app!

File: src/Main.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+function createMainWindow() {
+  mainWindow = new BrowserWindow({ width: 800, height: 600 });
+
+  mainWindow.loadURL(url.format({
+    pathname: path.join(__dirname, 'index.html'),
+    protocol: 'file:',
+    slashes: true
+  }));
+
+  mainWindow.webContents.openDevTools();
```
Comment (marinhero) [inline src/Main.ts]: Looks like `BrowserWindow` will be our favorite object

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{
+  "compilerOptions": {
```
Comment (marinhero) [inline tsconfig.json]: It's so nice to see compilation flags!

File: .gitignore [hunk 6]
```diff
@@ -1 +1,4 @@
+app/*
+!app/index.html
```
Comment (marinhero) [inline .gitignore]: Help me understand, why do we add a file that we want to ignore ?

File: package.json [hunk 7]
```diff
@@ -1,5 +1,27 @@
...
+  },
+  "bugs": {
+    "url": "https://github.com/fantasmas/rasa/issues"
+  },
+  "main": "./app/Main.js",
+  "scripts": {
+    "build-osx": "bin/build-osx",
+    "start": "bin/start"
+  },
   "devDependencies": {
```
Comment (marinhero) [inline package.json]: For some reason even if `electron-packager` is listed here my machine won't find the executable and will fail while running the build script. I'm sure it's a `PATH` issue... or did you installed through `npm -g` ?

File: src/Main.ts (hunk 3 above)
Comment (hachibu) [inline src/Main.ts]: Me too!

File: tsconfig.json (hunk 5 above)
Comment (hachibu) [inline tsconfig.json]: Yeah this is to configure the various levels of strictness in the TypeScript compiler.

File: .gitignore (hunk 6 above)
Comment (hachibu) [inline .gitignore]: This is a hack. index.html should be in src and copied to app via a gulp task.

File: package.json (hunk 7 above)
Comment (hachibu) [inline package.json]: You need to be using yarn for everything. No more npm or global stuff. Yarn is like bundle exec.

`yarn electron-packager`

Comment (hachibu) [inline package.json]: It helps keep everything local and versioned. No globals.

Comment (marinhero) [inline package.json]: Perfect! thanks

**Comment by marinhero** [comment]
screen shot 2016-12-28 at 11 28 12 am

Just made a working build by using yarn in the bundle exec fashion. This is good to go 👍

**Pull request merged**